user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementDetailsComponent.java

```diff
@@ -425,8 +448,8 @@ public void addListener(ComponentEventListener<MeasurementSelectionChangedEvent>
 
   //TODO introduce custom tab with label and updateable count
   public Optional<String> getSelectedTabName() {
-    if (registeredMeasurementsTabSheet.getSelectedTab() != null) {
-      return Optional.of(registeredMeasurementsTabSheet.getSelectedTab().getLabel());
+    if (registeredMeasurementsTabSheet.getSelectedTab() != null && !tabsInTabSheet.isEmpty()) {
+      return Optional.of(tabsInTabSheet.get(registeredMeasurementsTabSheet.getSelectedIndex()).getTabLabel());
```

Comment:
Why so complex?
Doesn't this method produce the same as
```java
Optional.ofNullable(registeredMeasurementsTabSheet.getSelectedTab()).map(Tab::getLabel);
```

Reply:
I tried it, but it doesn't work like that. Tabsheets are weird.

Reply:
Did for me 😄 I tried it too

Reply:
It didn't correctly update the selections for me and threw Nullpointer exceptions at one point

Reply:
it's not the same logic, as the item index is needed to get the correct tab. I can test it again to pinpoint the problem, if you think it's worthwhile

It does also not check if tabsInTabSheet is empty, which lead to the NPE iirc

Reply:
problem was the custom tab, which can't set the tab label. hope it's still better to read now, as I need to cast